federation/pkg/federation-controller/namespace/namespace_controller.go

```diff
@@ -366,27 +367,38 @@ func (nc *NamespaceController) reconcileNamespace(namespace string) {
 			ObjectMeta: util.DeepCopyRelevantObjectMeta(baseNamespace.ObjectMeta),
 			Spec:       *(util.DeepCopyApiTypeOrPanic(&baseNamespace.Spec).(*apiv1.NamespaceSpec)),
 		}
+
+		// Check to see if this should be sent to the cluster.
+		send, err := clusterselector.SendToCluster(cluster.Labels, desiredNamespace.ObjectMeta.Annotations)
+		if err != nil {
+			glog.Errorf("Error processing ClusterSelector cluster: %s for namespace: %s error: %s", cluster.Name, namespace, err.Error())
+			return
+		} else if !send {
+			glog.V(5).Infof("Skipping cluster: %s for namespace: %s reason: cluster selectors do not match: %-v %-v", cluster.Name, namespace, cluster.ObjectMeta.Labels, desiredNamespace.ObjectMeta.Annotations[federationapi.FederationClusterSelectorAnnotation])
+			continue
```

Comment:
This causes you to skip over the switch statement below, and in particular the "case found && !send" so you do not end up deleting the namespace in an underlying cluster when you should.